apps/vs-code-designer/src/app/commands/createLogicApp/createLogicAppSteps/LogicAppFileShareStep.ts

```diff
@@ -0,0 +1,114 @@
+import { type ShareClient, ShareServiceClient, StorageSharedKeyCredential } from '@azure/storage-file-share';
+import { AzureWizardExecuteStep, type IActionContext, callWithTelemetryAndErrorHandling, nonNullProp } from '@microsoft/vscode-azext-utils';
+import type { ILogicAppWizardContext } from '@microsoft/vscode-extension';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import type * as vscode from 'vscode';
+
+type File = {
+  path: string;
+  name: string;
+};
+
+export class LogicAppFileShareStep extends AzureWizardExecuteStep<ILogicAppWizardContext> {
+  public priority = 140;
+
+  public async execute(
+    wizardContext: ILogicAppWizardContext,
+    progress: vscode.Progress<{ message?: string; increment?: number }>
+  ): Promise<void> {
+    await callWithTelemetryAndErrorHandling('uploadFileShareStep', async (context: IActionContext) => {
+      context.telemetry.properties.storage = wizardContext.storageAccount?.name;
+      context.telemetry.properties.fileShare = wizardContext.newSiteName;
+
+      const message: string = localize('uploadingFiles', 'Uploading files to File Share "{0}"...', wizardContext.newSiteName);
+      ext.outputChannel.appendLog(message);
+      progress.report({ message });
+
+      const storageClient: StorageManagementClient = await createStorageClient(wizardContext);
+      const storageShareClient = await this.createStorageClient(wizardContext, storageClient);
+      const shareName = getNewFileShareName(nonNullProp(wizardContext, 'newSiteName'));
+      const shareClient = storageShareClient.getShareClient(shareName);
+
+      await this.createFileShare(shareClient);
+
+      const rootDirectories = [deploymentsDirectory, diagnosticsDirectory, locksDirectory, wwwrootDirectory];
+      await this.createDirectories(shareClient, rootDirectories);
+
+      const workspaceFolder = await getWorkspaceFolderPath(wizardContext);
+      const projectPath: string | undefined = await tryGetFunctionProjectRoot(wizardContext, workspaceFolder, true /* suppressPrompt */);
+      await this.uploadRootFiles(shareClient, projectPath);
+
+      await this.uploadWorkflowsFiles(shareClient, projectPath);
+    });
+  }
+
+  public shouldExecute(context: ILogicAppWizardContext): boolean {
+    return !!context.useContainerApps;
+  }
+
+  private async createStorageClient(context: ILogicAppWizardContext, client: StorageManagementClient): Promise<ShareServiceClient> {
+    const keys: StorageAccountListKeysResult = await client.storageAccounts.listKeys(
+      context.resourceGroup.name,
+      context.storageAccount.name
+    );
+    const credential = new StorageSharedKeyCredential(context.storageAccount.name, keys.keys[0].value);
+    return new ShareServiceClient(`https://${context.storageAccount.name}.file.core.windows.net`, credential);
+  }
+
+  private async createFileShare(shareClient: ShareClient) {
+    await shareClient.createIfNotExists();
```

Comment:
If we create from the ARM call the root file structure is going to be the same  as this one
<img width="1635" alt="image" src="https://github.com/Azure/LogicAppsUX/assets/102700317/fa557b99-bfe6-4c95-a458-7ce8f3821459">

Instead of this one, you mentioned me is the desired one
<img width="1650" alt="image" src="https://github.com/Azure/LogicAppsUX/assets/102700317/f2d8f80f-6049-4b28-98e8-84b3159c9134">

Unless BE handles it

Reply:
Yes, that BE changes are in progress so it will be handled